src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -1227,7 +1227,7 @@ private bool IsCommandMatch(ref CommandInfo current, out bool isDuplicate)
 
                 if (isCommandMatch)
                 {
-                    if (ArgumentList != null)
+                    if ((current.CommandType == CommandTypes.Alias) && Syntax)
```

Comment:
General question for more knowledgeable people, would it be faster this way or  with the conditions flipped (`(Syntax && (current.CommandType == CommandTypes.Alias))`)? 

This'll run on every Get-Command call (that finds a match) so I feel like having them the other way around would be better since the number of times Syntax is specified is going to be much smaller so it'll fail fast through the condition. If someone can point me to a good guide on how to actually compare them then that'd be useful too. I'd guess the difference is minimal but if one is faster than the other then it makes more sense to use whichever it is.

Reply:
If the condition is `&&` there isn't really a "positive" short-circuit since both conditions have to be satisfied, but there may be good reason to negatively short-circuit the code path. If there is a condition which is more commonly `false` than the other one, that should be first.

Reply:
Also, I think it's probably a good idea to use `Syntax.IsPresent` so the compiler doesn't need to bother invoking the implicit cast to boolean behind the scenes.

Reply:
For me `Syntax.IsPresent && (current.CommandType == CommandTypes.Alias` is more readable.

Reply:
Fixed.

Reply:
That's a much cleaner solution. I've added that now.